src/grammar/grammar.ohm

```diff
@@ -143,6 +144,8 @@ Tact {
 
     StatementForEach = foreach "(" id "," id "in" Expression ")" "{" Statement* "}"
 
+    StatementDestruct = let "{" ListOf<id, ","> ","? "}" "=" Expression (";" | &"}")
```

Comment:
We need to use an order-independent approach for struct fields, like this:

```tact
struct S {
  field1: Int;
  field2: Int;
}

let S { field1: a, field2: b } = struct;
// `a` and `b` are now bound to the `field1` and `field2` components respectively
```

and also using struct field punning should be allowed too:
```tact
let S { field1, field2 } = struct;
// `field1` and `field2` are bound to the respective struct fields
```

Notice that in the second example the order of fields is not relevant.

This prevents issues with reordering of struct fields in the struct definitions. The currently implemented design would provoke subtle bugs if the user changes the order of fields like so:

```
struct S {
  field2: Int;
  field1: Int;
}
```

Since the fields have the same type the refactoring which fails to change the order in the destructuring let-statements isn't going to be frowned upon by the typechecker.